### ISSUE TITLE
Blogpost on how you programmed this game?

### ISSUE THREAD

**Issue by ghost** [loov/zombieroom#2]
Hi loov,
i would like to know learn about how you started creating this game with Go.
And how you figured out the input.go file since its working really nice with my SNES USB Controller.

Is there a blog post about this our could I write you a mail?

**Comment by egonelbre** [comment]
Name is actually Egon -- I just keep more "complete projects" in a "separate organization".

How it started? I'm a mentor on local game jam events -- so when people have difficult problems or don't know how to do X, I'll help them. Sometimes I participate with a team, but during LD37 I decided to take it easier and not participate. Effectively, I got somewhat bored, so I ended up writing this game in the pauses when I wasn't helping anyone. I've written this game mechanic multiple times, so I already knew what needed to be done.

As for the `input.go`. I wanted the game to be such that people can drop in and out of the game as they want (supports 6 players at the moment), and can use controller or keyboard. Now, dealing at "game logic level" with interfaces for input is quite annoying... So instead I decided to create a separate "Controller" to normalize input into a single struct. So, for example if you wanted to make some other game, you would modify the "Controller" and then the specific "Updater" will update it on each frame depending on the device. For the SNES USB Controller, I read it using `glfw`, which handles bulk of the complexity.

As for the discussion, I think the best place at the moment is channel #gamedev in Gophers Slack. I can discuss things there and there are plenty of other people who can jump in there as well.
